engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/job/ExecuteExpressionJob.java

```diff
@@ -0,0 +1,49 @@
+package org.enso.interpreter.instrument.job;
+
+import com.oracle.truffle.api.TruffleLogger;
+import java.util.UUID;
+import java.util.logging.Level;
+import org.enso.interpreter.instrument.Visualization;
+import org.enso.interpreter.instrument.execution.Executable;
+import org.enso.interpreter.instrument.execution.RuntimeContext;
+import org.enso.interpreter.util.ScalaConversions;
+
+public class ExecuteExpressionJob extends Job<Executable> {
+
+  private final UUID contextId;
+  private final UUID visualizationId;
+  private final UUID expressionId;
+  private final String expression;
+
+  public ExecuteExpressionJob(
+      UUID contextId, UUID visualizationId, UUID expressionId, String expression) {
+    super(ScalaConversions.cons(contextId, ScalaConversions.nil()), false, false);
+    this.contextId = contextId;
+    this.visualizationId = visualizationId;
+    this.expressionId = expressionId;
+    this.expression = expression;
+  }
+
+  @Override
+  public Executable run(RuntimeContext ctx) {
+    TruffleLogger logger = ctx.executionService().getLogger();
+    long lockTimestamp = ctx.locking().acquireContextLock(contextId);
+
+    try {
+      Visualization visualization =
+          new Visualization.OneshotExpression(visualizationId, expressionId, contextId, expression);
+      ctx.contextManager().upsertVisualization(contextId, visualization);
+
+      var stack = ctx.contextManager().getStack(contextId);
+      return new Executable(contextId, stack);
```

Comment:
Executable is `contextId` and ``Stack[InstrumentFrame]` - who's going to consume the return value?

Reply:
After the visualization is set up, we run the program to inspect the node. The executable will be used to schedule an `ExecuteJob`